Add info about new legacy browser filters (#9096)
getsentry/sentry-docs: docs/product/data-management-settings/filtering/index.mdx

- Common browser extension errors
- Events coming from localhost
- From legacy browsers
- Known web crawlers
- By their error message
- From specific release versions of your code
- From certain IP addresses
- React hydration errors

> - Transactions coming from healthcheck and ping requests

Reviewer: I would reorder these to follow the same order as the UI:

- Common browser extension errors
- Transactions coming from health checks and ping requests
- Events coming from legacy browsers
- Events coming from localhost
- Errors from known web crawlers
- React hydration errors
- ChunkLoadErrors
- Events from certain IP addresses
- Events with certain error messages
- Events from specific release versions of your code

### Legacy Browser Filters

The legacy browser filters allow you to filter out certain legacy versions of browsers that are known to cause problems. 

> As of February 2024, we have new, updated legacy browser filters, that are more up to date than the previous ones. These new filters will be periodically evaluated and updated to filter out versions of the browsers that we believe should be considered legacy. 

Reviewer: ```suggestion
Legacy browser filters were updated in Feb 2024 and will be periodically evaluated to include additional legacy versions. 
```

> Any old legacy browser filters will still show up in Settings, but will be labeled with "Deprecated". They will continue to work, but you will not be able to turn them on if you don't already have them on. Once you turn an old filter off, you will be unable to turn it on again. 

Reviewer: ```suggestion
If you had a legacy browser filter on before the update, the old filter will appear in your settings as "Deprecated". Deprecated legacy browser filters still work. However, if you turn them off, you won’t be able to turn them on again and will need to use the new filters instead.```

> ### Transactions coming from healthcheck

Reviewer: ```suggestion
### Transactions Coming from Health Check
```
